test/controllers/SummaryDeclarationControllerSpec.scala

```diff
@@ -88,235 +82,510 @@ class SummaryDeclarationControllerSpec
+    "show the summary declaration page" in {
       val authRequest = buildRequestWithAuth(Fixtures.buildFakeRequestWithSessionId("GET"))
 
-      val result = controllerUnderTest.showSummaryDeclarationPage(ersRequestObject)(authRequest)
-      status(result) shouldBe Status.OK
+      val findList = Seq(
+        "scheme-type", "portal-scheme-ref", "scheme-organiser", "group-scheme-controller", "subsidiary-companies", "trustees", "reportable-events", "ErsMetaData"
+      )
+      val addList = Map(
+        "check-file-type" -> Json.toJson(fileTypeODS),
+        "file-name"       -> Json.toJson(fileNameODS),
+        "alt-activity"    -> Json.toJson(new AltAmendsActivity("2"))
+      )
+
+      when(mockErsUtil.buildEntitySummary(any())).thenReturn("Company Name, Add1, Add2, Add3, Add4, UK, AA111AA, AB123456, 1234567890")
+      when(mockSessionService.fetchAll()(any())).thenReturn(Future.successful(setCacheItem("id1", findList, addList)))
+
+      val result = testSummaryDeclarationController.showSummaryDeclarationPage(ersRequestObject)(authRequest)
+
+      status(result) shouldBe OK
+      val document = Jsoup.parse(contentAsString(result))
+      document.getElementsByTag("title").get(0).text shouldBe Messages("ers.summary.page_title")
     }
   }
 
   "Calling SummaryDeclarationController.showSummaryDeclarationPage (GET) with authentication and required elements (no group scheme info) in the cache" should {
-    "show the scheme organiser page" in {
-      val controllerUnderTest = buildFakeSummaryDeclarationController("noGroupSchemeInfo")
+    "show the error page if group scheme info is missing" in {
       val authRequest = buildRequestWithAuth(Fixtures.buildFakeRequestWithSessionId("GET"))
 
-      val result = controllerUnderTest.showSummaryDeclarationPage(ersRequestObject)(authRequest)
-      status(result) shouldBe Status.OK
+      val findList =
+        Seq("scheme-organiser", "subsidiary-companies", "trustees", "ErsMetaData")
+      val addList =
+        Map(
+          "reportable-events" -> Json.toJson(new ReportableEvents(Some(OPTION_NIL_RETURN))),
+          "alt-activity"      -> Json.toJson(new AltAmendsActivity("2"))
+        )
+
+      when(mockErsUtil.buildEntitySummary(any())).thenReturn("Company Name, Add1, Add2, Add3, Add4, UK, AA111AA, AB123456, 1234567890")
+      when(mockSessionService.fetchAll()(any())).thenReturn(Future.successful(setCacheItem("id1", findList, addList)))
+
+      val result = testSummaryDeclarationController.showSummaryDeclarationPage(ersRequestObject)(authRequest)
+
+      status(result) shouldBe INTERNAL_SERVER_ERROR
+      val document = Jsoup.parse(contentAsString(result))
+      document.getElementsByTag("title").get(0).text shouldBe Messages("ers.global_errors.title")
+    }
+  }
+
+  "Calling SummaryDeclarationController.showSummaryDeclarationPage (GET) with authentication and group scheme set to 'Yes" should {
```

Comment:
there are a lot of new tests in this file, they all look quite similar can we make a list with input and outputs like in `test/controllers/GroupSchemeControllerSpec.scala` (see comments) to cut down the amount of code reuse?